components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -598,17 +601,23 @@ void AdsServiceImpl::OnInitialize(
     const int32_t result) {
   if (result != ads::Result::SUCCESS) {
     LOG(ERROR) << "Failed to initialize ads";
-    return;
+    is_initialized_ = false;
   }
+  else {
+    is_initialized_ = true;
 
-  is_initialized_ = true;
+    // OnBraveAdsServiceReady is implemented for Android for now
+    #if defined(OS_ANDROID)
```

Comment:
`NotificationDisplayServiceImpl::OnBraveAdsServiceReady` is a callback function implemented in `browser-android-tabs`  for now. Potentially it might be used for all platforms.  Wiring 2 lines of code through the above pattern will  add a number of files we have to support. I would leave it as it is. Now it's clearly and explicitly says: `this code only runs for Android`!

Reply:
By adding platform specific code for different platforms adds unneeded bloat the service layer. Platforms which do not currently have a need can just use the default implementation keeping the files to a minimum but allowing change as and when needed on a per platform basis

Reply:
@tmancey again it is your point of view and how you prefer it to see. Please do `git grep "#if defined(OS_ANDROID)"` inside chromium's src folder and check how does chromium do it.

Reply:
@SergeyZhukovsky We have `notification_helper_android.cc` (see above) along with implementations for macOS, linux and windows where code for platform specific notification code resides. By adding to the service layer, the class is just getting bigger and bigger when we should be trying to keep classes to a reasonable size (and I know ads service layer is not perfect but we are gradually decoupling the code)

Reply:
Issue created for that https://github.com/brave/brave-browser/issues/6200